src/vs-bicep/Bicep.VSLanguageServerClient/MiddleLayerProviders/UpdateFormatSettingsMiddleLayer.cs

```diff
@@ -34,21 +42,33 @@ public async Task HandleNotificationAsync(string methodName, JToken methodParam,
         {
             if (CanHandle(methodName))
             {
-                methodParam = UpdateFormatOptions(methodParam);
+                methodParam = await UpdateFormatOptionsAsync(methodParam);
             }
 
             return await sendRequest(methodParam);
         }
 
-        public JToken UpdateFormatOptions(JToken methodParam)
+        public async Task<JToken> UpdateFormatOptionsAsync(JToken methodParam)
         {
             var documentFormattingParams = methodParam.ToObject<DocumentFormattingParams>();
 
             if (documentFormattingParams is not null)
             {
                 var formattingOptions = documentFormattingParams.Options;
-                formattingOptions.InsertSpaces = true;
-                formattingOptions.TabSize = 2;
+
+                var formatterIndentTypeKey = await bicepSettings.GetIntegerAsync(BicepLanguageServerClientConstants.FormatterIndentTypeKey, (int)IndentType.Spaces);
+
+                if (formatterIndentTypeKey == (int)IndentType.Tabs)
+                {
+                    formattingOptions.InsertSpaces = false;
+                    formattingOptions.TabSize = await bicepSettings.GetIntegerAsync(BicepLanguageServerClientConstants.FormatterTabSizeKey, 2);
```

Comment:
I'd create constants for these two defaults

Reply:
I currently don't detect settings changes. So this needs to be fetched when formatting is invoked. Right approach here would be to detect settings change, cache settings information to a writable storage and use that when formatting is invoked. It's requires lot more work and I created https://github.com/Azure/bicep/issues/8218 to track it. We can revisit it if we get feedback from customers. Let me know if you have any concerns. Thanks!